src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -1585,7 +1597,134 @@ else if (ticket == TicketType.POST_TELEPORT) // post teleport
                 cc.s.getLevel().getChunkSource().addRegionTicket(TicketType.UNKNOWN, target, radius, target);
             return new NumericValue(ticket.timeout());
         });
+        // =======================================
+        expression.addContextFunction("save_structure_template", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerLevel world = cc.s.getLevel();
+
+            StructureManager structureManager = world.getStructureManager();
+            StructureTemplate theStructure;
+            String name;
+
+            boolean returnnbt = lv.get(0).isNull();
+            ResourceLocation struident = null;
+            if (!returnnbt) {
+                name = lv.get(0).getString();
+                struident = ResourceLocation.tryParse(name);
+                if (name.isEmpty() || name.endsWith(":") || struident == null) {
```

Comment:
seems like there is more to proper validation of structure names. maybe some convention should be used here that is acceptable on the client?

Reply:
there are more checks in 'tryParse'.
i think that is enough.

I didn't do any more validation at all at first and just used `tryParse`. but was surprised to find out during testing that the path is empty actually bypasses this function's check. That's how I added the rule that it can't end with a colon.

Reply:
use InputValidator::identifierOf since its throws proper exception without you needing to handle it. 

If you have found some extra usecases, you can update InputValidator::identifierOf to include these extra cases

Reply:
> use InputValidator::identifierOf since its throws proper exception without you needing to handle it.
> 
> If you have found some extra usecases, you can update InputValidator::identifierOf to include these extra cases

ok. i start using identifierOf and add a small check in identifierOf.